### PR TITLE
Port the collisions_updater to ROS2

### PR DESCRIPTION
**Pull request by brennand**
Description

This is a port for the collision updater:

1) Collision updater 100% working in ROS2
2) removed the launch file as its for ROS1 and it launches the GUI which hasn't been ported yet.  I decided to remove it so that people are not confused and try and run it.
3) Added comment in the README of how to run the command line tool
4) Striped all the GUI "stuff" out of the CMake as it will have to be re-written when the GUI is ported.
5) changed from .htp .hpp as that seems to be the ROS2 way?

linked against this issue:

https://github.com/ros-planning/moveit2/issues/724

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: moveit_setup_assistant/CMakeLists.txt [hunk 1]
```diff
@@ -1,169 +1,91 @@
...
-  src/widgets/author_information_widget.cpp
-  ${HEADERS}
+ament_target_dependencies(
+  ${PROJECT_NAME}_tools
+  "moveit_core"
+  "moveit_ros_planning"
+  "moveit_ros_visualization"
+  "rclcpp"
+  "urdf"
+  "srdfdom"
```
Comment (tylerjw) [inline moveit_setup_assistant/CMakeLists.txt:36]: you shouldn't need quotes here

File: moveit_setup_assistant/CMakeLists.txt [hunk 2]
```diff
@@ -1,169 +1,91 @@
...
-  ${PROJECT_NAME}_widgets ${PROJECT_NAME}_tools
-  ${QT_LIBRARIES} ${catkin_LIBRARIES} ${Boost_LIBRARIES} log4cxx
+ament_target_dependencies(
+  ${PROJECT_NAME}_updater
+  "moveit_core"
+  "moveit_ros_planning"
+  "moveit_ros_visualization"
+  "rclcpp"
+  "urdf"
+  "srdfdom"
```
Comment (tylerjw) [inline moveit_setup_assistant/CMakeLists.txt:55]: no need for quotes here either

File: moveit_setup_assistant/CMakeLists.txt [hunk 3]
```diff
@@ -1,169 +1,91 @@
...
+)
 
-install(TARGETS ${PROJECT_NAME} ${PROJECT_NAME}_updater
-  RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION})
+install(
+  TARGETS
+    ${PROJECT_NAME}_updater
+  DESTINATION
+    lib/${PROJECT_NAME}
+)
```
Comment (tylerjw) [inline moveit_setup_assistant/CMakeLists.txt:69]:
```suggestion
install(
  TARGETS
    ${PROJECT_NAME}_tools
    ${PROJECT_NAME}_updater
  EXPORT ${PROJECT_NAME}Targets
  ARCHIVE DESTINATION lib
  LIBRARY DESTINATION lib
  RUNTIME DESTINATION lib/${PROJECT_NAME}
  INCLUDES DESTINATION include
)
install(DIRECTORY launch DESTINATION share/${PROJECT_NAME})
install(DIRECTORY resources DESTINATION share/${PROJECT_NAME})
install(DIRECTORY templates DESTINATION share/${PROJECT_NAME})
ament_export_targets(${PROJECT_NAME}Targets HAS_LIBRARY_TARGET)
ament_export_dependencies(${THIS_PACKAGE_INCLUDE_DEPENDS})
```

from my other review...

File: moveit_setup_assistant/CMakeLists.txt [hunk 4]
```diff
@@ -1,169 +1,91 @@
...
-install(TARGETS ${PROJECT_NAME}_widgets ${PROJECT_NAME}_tools
-  LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  RUNTIME DESTINATION ${CATKIN_GLOBAL_BIN_DESTINATION})
+install(TARGETS ${PROJECT_NAME}_tools
+  EXPORT export_${PROJECT_NAME}
+  ARCHIVE DESTINATION lib
+  LIBRARY DESTINATION lib
+  RUNTIME DESTINATION bin
+)
```
Comment (tylerjw) [inline moveit_setup_assistant/CMakeLists.txt:76]:
```suggestion
```

File: moveit_setup_assistant/launch/setup_assistant.launch [hunk 5]
```diff
@@ -1,14 +0,0 @@
...
-  <arg name="debug" default="false" />
-  <arg unless="$(arg debug)" name="launch_prefix" value="" />
-  <arg     if="$(arg debug)" name="launch_prefix" value="gdb --ex run --args" />
-
-  <!-- Run -->
-  <node pkg="moveit_setup_assistant" type="moveit_setup_assistant" name="moveit_setup_assistant"
-        launch-prefix="$(arg launch_prefix)"
-        output="screen"  required="true" />
-
-</launch>
```
Comment (tylerjw) [inline moveit_setup_assistant/launch/setup_assistant.launch:14]: Why did you delete this file?  It is not part of your port.

File: moveit_setup_assistant/CMakeLists.txt [hunk 6]
```diff
@@ -1,169 +1,91 @@
...
-  src/widgets/configuration_files_widget.cpp
-  src/widgets/ros_controllers_widget.cpp
-  src/widgets/controller_edit_widget.cpp
-  src/widgets/navigation_widget.cpp
-  src/widgets/header_widget.cpp
-  src/widgets/setup_assistant_widget.cpp
-  src/widgets/setup_screen_widget.cpp
-  src/widgets/simulation_widget.cpp
-  src/widgets/author_information_widget.cpp
-  ${HEADERS}
```
Comment (tylerjw) [inline moveit_setup_assistant/CMakeLists.txt:127]: Comment out these lines, don't delete them, we still want to port them after your PR

File: moveit_setup_assistant/launch/setup_assistant.launch (hunk 5 above)
Comment (brennand) [reply moveit_setup_assistant/launch/setup_assistant.launch:14]: So that people don't try and run it.  I also thought that its such a simple launch file, we do not loss any useful information.

Comment (tylerjw) [reply moveit_setup_assistant/launch/setup_assistant.launch:14]: you can't run a .launch file in ros2, also this file is not part of your port

File: moveit_setup_assistant/CMakeLists.txt [hunk 7]
```diff
@@ -1,169 +1,142 @@
...
+  ARCHIVE DESTINATION lib
+  LIBRARY DESTINATION lib
+  RUNTIME DESTINATION lib/${PROJECT_NAME}
+  INCLUDES DESTINATION include
+)
+install(DIRECTORY launch DESTINATION share/${PROJECT_NAME})
+install(DIRECTORY resources DESTINATION share/${PROJECT_NAME})
+install(DIRECTORY templates DESTINATION share/${PROJECT_NAME})
+ament_export_targets(${PROJECT_NAME}Targets HAS_LIBRARY_TARGET)
+ament_export_dependencies(${THIS_PACKAGE_INCLUDE_DEPENDS})
```
Comment (tylerjw) [inline moveit_setup_assistant/CMakeLists.txt:127]: Did you define this somewhere?  I don't see it.  Look at the one in moveit_core for an example.

File: moveit_setup_assistant/src/tools/moveit_config_data.cpp [hunk 8]
```diff
@@ -1908,9 +1909,11 @@ srdf::Model::Group* MoveItConfigData::findGroupByName(const std::string& name)
 
   // Check if subgroup was found
   if (searched_group == nullptr)  // not found
-    ROS_FATAL_STREAM("An internal error has occured while searching for groups. Group '" << name
-                                                                                         << "' was not found "
-                                                                                            "in the SRDF.");
+  {
+    RCLCPP_ERROR_STREAM(LOGGER, "An internal error has occurred while searching for groups. Group '"
+                                    << name << "' was not found  in the SRDF.");
+    exit(-1);
```
Comment (tylerjw) [inline moveit_setup_assistant/src/tools/moveit_config_data.cpp:1915]: should this throw instead of calling exit?

Comment (brennand) [reply moveit_setup_assistant/src/tools/moveit_config_data.cpp:1915]: I could just remove it, to be a straight port.  This was written by Dlu, I guess the thinking, its better to exit here then carry on?

File: moveit_setup_assistant/CMakeLists.txt (hunk 7 above)
Comment (brennand) [reply moveit_setup_assistant/CMakeLists.txt:127]: This was a stright up copy/paste from your previous comment.

Comment (brennand) [reply moveit_setup_assistant/CMakeLists.txt:135]: Add the list

File: moveit_setup_assistant/src/tools/moveit_config_data.cpp (hunk 8 above)
Comment (tylerjw) [reply moveit_setup_assistant/src/tools/moveit_config_data.cpp:1915]:
```suggestion
    throw std::runtime_error(name + " was not found in the SRDF");
```
